lib/services/presets.go

```diff
@@ -877,6 +923,63 @@ func AddRoleDefaults(role types.Role) (types.Role, error) {
 	return role, nil
 }
 
+func mergeStrings(dst, src []string, dirty *bool) []string {
```

Comment:
This is very confusing function signature, can we have it return whether the role changed or not (boolean argument) rather than modify the passed in pointer?